homework/index.js

```diff
@@ -29,19 +31,162 @@
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
-    });
+  // clear all container content
+  function removeChildren(element) {
+    while (element.firstChild) {
+      element.removeChild(element.firstChild);
+    }
+  }
+
+  function renderError(error) {
+    const container = document.getElementById('container');
+    removeChildren(container);
+    // Render the error message in container
+    createAndAppend('div', container, { text: error.message, class: 'alert-error' });
   }
+  /* cSpell: disable */
+  const repoInfo = async selectedRepository => {
+    const container = document.getElementById('container');
 
+    // left div
+    const repoContainer = createAndAppend('div', container, {
+      class: 'left-div whiteframe',
+    });
+    // create a table
+    const table = createAndAppend('table', repoContainer);
+    const tbody = createAndAppend('tbody', table);
+
+    // table rows
+    // 1
+    const repository = createAndAppend('tr', tbody);
+    // 2
+    const description = createAndAppend('tr', tbody);
+    // 3
+    const forks = createAndAppend('tr', tbody);
+    // 4
+    const updated = createAndAppend('tr', tbody);
+
+    // table data
+    // 1
+    createAndAppend('td', repository, {
+      text: 'Repository :',
+      class: 'label',
+    });
+    const dataLink = createAndAppend('td', repository);
+    createAndAppend('a', dataLink, {
+      href: selectedRepository.html_url,
+      text: selectedRepository.name,
+      target: '_blank',
+    });
+    // 2
+    createAndAppend('td', description, {
+      text: 'Description :',
+      class: 'label',
+    });
+    createAndAppend('td', description, {
+      text: selectedRepository.description,
+    });
+    // 3
+    createAndAppend('td', forks, {
+      text: 'Forks :',
+      class: 'label',
+    });
+    createAndAppend('td', forks, {
+      text: selectedRepository.forks,
+    });
+    // 4
+    createAndAppend('td', updated, {
+      text: 'Updated :',
+      class: 'label',
+    });
+    createAndAppend('td', updated, {
+      text: selectedRepository.updated_at,
+    });
+
+    // right div
+    const contributorsHeader = createAndAppend('div', container, {
+      class: 'right-div whiteframe',
+    });
+    createAndAppend('p', contributorsHeader, {
+      text: 'contributions',
+      class: 'contributor-header',
+    });
+
+    // list
+    const ul = createAndAppend('ul', contributorsHeader, {
+      class: 'contributor-list',
+    });
+
+    // list items & data
+    const contributors = data => {
+      data.forEach(user => {
+        const li = createAndAppend('li', ul, {
+          class: 'contributor-item',
+        });
+        li.addEventListener('click', () => {
+          window.open(user.html_url, '_blank');
+        });
+
+        createAndAppend('img', li, { src: user.avatar_url, class: 'contributor-avatar' });
```

Comment:
There is one thing missing here which is mentioned as part of homework - `Make your app ARIA-compliant (see below).`. If you notice your app, it isn't accessible using the keyboard. Only the repo name(anchor link) is accessible using keyboard. One thing is that the contributors information isn't added as an anchor link but an event listener is being added for capturing any click events. Could you try to add an anchor link using which the browser would automatically add it as part of keyboard navigation.

Reply:
I hope it works now